### DISCUSSION TITLE
Error fetching service information from Docker: Error response from daemon: This node is not a swarm manager. Use "docker swarm init" or "docker swarm join" to connect this node to swarm and try again.

### DISCUSSION THREAD

**Discussion by sbnair** [pulumi/pulumi#9428] [Pulumi in Practice]
When i am trying to run docker using pulumi import command i get this error : 
Error fetching service information from Docker: Error response from daemon: This node is not a swarm manager. Use "docker swarm init" or "docker swarm join" to connect this node to swarm and try again.

Please let me know what i need to do ?

**Comment by rawkode** [comment]
Services are only available when the Docker Engine is configured to run as an orchestrator, which is called Docker Swarm (or Swarm Mode)

https://docs.docker.com/engine/swarm/

**Reply by sbnair** [reply]
So if i need to use pulumi to run a docker node using import command i need to configure docker swarm ?

**Comment by rawkode** [comment]
No, if you want to deploy a Docker Service. Perhaps you could share your Pulumi code or command you’re running to let us help you further.

**Reply by sbnair** [reply]
pulumi import docker:index/service:Service mina a517385c7bd3
This is the code that i am running and mina is the name of the docker and the last one is the container id, pls let me know what needs to be done ?.

**Reply by sbnair** [reply]
I want to run this docker command using pulumi command, is it possible ?
sudo docker run --name mina -d -p 8302:8302 --restart=always --mount "type=bind,source=$(pwd)/.mina-env,dst=/entrypoint.d/mina-env,readonly" --mount "type=bind,source=$(pwd)/keys,dst=/keys,readonly" --mount "type=bind,source=$(pwd)/.mina-config,dst=/root/.mina-config" minaprotocol/mina-daemon:1.3.0-9b0369c-bullseye-mainnet daemon

**Reply by rawkode** [reply]
Yes, use the `container` API and not the service API. The service API is **ONLY for Swarm**.

You can run containers with the following docs:

https://www.pulumi.com/registry/packages/docker/api-docs/container/

**Reply by sbnair** [reply]
Can you share some examples which has converted the existing docker into pulumi, i don't find much in the docs ?

**Reply by sbnair** [reply]
because i need to pass too many mounting variables into this docker based on the command i have given above, so if there are working docker examples it will be of great help.

**Comment by sbnair** [comment]
When i try to use the example that you gave me using container API i am getting this error : 
    error obtaining VCS status: exit status 128
    	Use -buildvcs=false to disable VCS stamping.
 
    error: an unhandled error occurred: program exited with non-zero exit code: 1
